lighthouse-core/lib/network-recorder.js

```diff
@@ -57,10 +57,21 @@ class NetworkRecorder extends EventEmitter {
     return this._isActiveIdlePeriod(0);
   }
 
+  /**
+   * Returns whether any important resources for the page are in progress.
+   * Above-the-fold images and XHRs should be included.
+   * Tracking pixels, low priority images, and cross frame requests should be excluded.
+   * @return {boolean}
+   */
   isCriticalIdle() {
+    const rootFrameRequest = this._records.find(r => r.resourceType === 'Document');
+    const rootFrameId = rootFrameRequest && rootFrameRequest.frameId;
+    const rootFrameRequests = new Set(this._records.filter(r => r.frameId === rootFrameId));
+
     return this._isActiveIdlePeriod(
       0,
-      request => request.priority === 'VeryHigh' || request.priority === 'High'
+      request => rootFrameRequests.has(request) &&
```

Comment:
could this be
```suggestion
      request => request.frameId === rootFrameId &&
```
and avoid the `Set` construction/`.has()` call?

Reply:
🤦 it sure could